testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -14,29 +14,190 @@
+                        -1000000000000012L, // a_bigint
+                        -10000000.123F, // a_real
+                        -100000000000.123, // a_double
+                        new BigDecimal("-1234567.8901"), // a_short_decimal
+                        new BigDecimal("-1234567890123456789.0123456"), // a_long_decimal
+                        "", // a_string
+                        java.sql.Date.valueOf(LocalDate.of(1965, 9, 10)), // a_date
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0)), // a_timestamp_seconds
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0, 123_000_000)), // a_timestamp_millis
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_micros
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_nanos TODO/note nanos were truncated
+                        "dummy"));
+        assertThat(onSpark().executeQuery("SELECT * FROM " + sparkTableName)).containsOnly(expected);
+        onTrino().executeQuery("SET SESSION hive.timestamp_precision = 'NANOSECONDS'");
+        assertThat(onTrino().executeQuery("SELECT * FROM " + trinoTableName)).containsOnly(expected);
+
+        assertThat(onTrino().executeQuery("SHOW CREATE TABLE " + trinoTableName))
+                .containsOnly(row(format(
+                        "CREATE TABLE %s (\n" +
+                                "   a_boolean boolean,\n" +
+                                "   a_tinyint tinyint,\n" +
+                                "   a_smallint smallint,\n" +
+                                "   an_integer integer,\n" +
+                                "   a_bigint bigint,\n" +
+                                "   a_real real,\n" +
+                                "   a_double double,\n" +
+                                "   a_short_decimal decimal(11, 4),\n" +
+                                "   a_long_decimal decimal(26, 7),\n" +
+                                "   a_string varchar,\n" +
+                                "   a_date date,\n" +
+                                "   a_timestamp_seconds timestamp(9),\n" +
+                                "   a_timestamp_millis timestamp(9),\n" +
+                                "   a_timestamp_micros timestamp(9),\n" +
+                                "   a_timestamp_nanos timestamp(9),\n" +
+                                "   a_dummy varchar\n" +
+                                ")\n" +
+                                "WITH (\n" +
+                                "   format = '%s'\n" +
+                                ")",
+                        trinoTableName,
+                        expectedTrinoTableFormat)));
+
+        onSpark().executeQuery("DROP TABLE " + sparkTableName);
+    }
+
+    @DataProvider
+    public static Object[][] testReadSparkCreatedTableDataProvider()
```

Comment:
With just two values in data provider I would rather have separate test methods for ORC and Parquet. Make manual running slightly easier. Also data providers sometimes make log output confusing.

Reply:
i think we should add more cases here. like avro.